CDCgov/prime-reportstream
# 11549: NTE HL7 -> FHIR -> HL7

This PR verifies and updates that ReportStream can translate NTE segments to FHIR and back to HL7 with no modifications to the original segment

Test Steps:
1. Running the tests
2. Confirm that the FHIR inventory and all the NIST 2.5.1 fields are covered by the mappings
  - https://docs.google.com/spreadsheets/d/1ycH9LSI0xRyZqiRmh233wdpmJ-MyjzPFe4015X2TDDY/edit#gid=0
  - The definition of NTE here https://hl7v2-elr-testing.nist.gov/mu-elr/

## Changes
- Update the NTE -> Annotation
- Update Annoation -> NTE

## Checklist

### Testing

### Process

## Linked Issues
- #11549 

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
